### PR TITLE
Cleanup project structure and configuration

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -27,66 +27,76 @@ plugins {
...
+
+val coroutines_version: String by extra
+
+kotlin {
+    jvm {
+        compilations.all {
+            kotlinOptions.jvmTarget = "1.8"
+        }
+        testRuns["test"].executionTask.configure {
+            useJUnitPlatform()
```
Comment (whyoleg) [inline build.gradle.kts:50]: isn't it default configuraiton?

File: gradle.properties [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+kotlin.mpp.stability.nowarn=true
+kotlin.mpp.enableCompatibilityMetadataVariant=true
+kotlin.mpp.enableCInteropCommonization=true
+kotlin.internal.mpp.hierarchicalStructureByDefault=true
 
 # versions
 coroutines_version=1.6.0
 atomicfu_version=0.17.0
-kotlin_version=1.6.10
+kotlin_version=1.7.0
```
Comment (whyoleg) [inline gradle.properties:16]: minor: I prefer version catalogs

File: settings.gradle.kts [hunk 3]
```diff
@@ -1,4 +1 @@
-
 rootProject.name = "ktor-io"
-
-include(":core")
```
Comment (whyoleg) [inline settings.gradle.kts:4]: Not sure, if it's a good change, as I believe 1 module will be not enough for `io` library :)

**Review by whyoleg — APPROVED** [review]

File: build.gradle.kts [hunk 4]
```diff
@@ -27,66 +27,76 @@ plugins {
...
-        jvm {
-            compilations.all {
-                kotlinOptions.jvmTarget = "1.8"
-            }
-            withJava()
-            testRuns["test"].executionTask.configure {
-                useJUnitPlatform()
+    sourceSets {
+        val commonMain by getting {
+            dependencies {
```
Comment (rsinukov) [inline build.gradle.kts:66]: why do you need an empty block here?

**Review by rsinukov — APPROVED** [review]

File: build.gradle.kts [hunk 5]
```diff
@@ -27,66 +27,76 @@ plugins {
...
+    js(IR){
+        browser()
+        nodejs()
     }
 
-    apply(plugin = "kotlin-multiplatform")
-    apply(plugin = "kotlinx-atomicfu")
+    mingwX64()
+    linuxX64()
+    macosX64()
```
[the remark this answers is not in the document]
Comment (e5l) [reply build.gradle.kts:60]: Done

File: gradle.properties (hunk 2 above)
Comment (e5l) [reply gradle.properties:16]: will migrate later, will be hard to rebase now

File: settings.gradle.kts (hunk 3 above)
Comment (e5l) [reply settings.gradle.kts:4]: Let's not introduce complexity in advance, will add later when it will be necessary

File: build.gradle.kts (hunk 4 above)
Comment (e5l) [reply build.gradle.kts:66]: Thanks, dropped!

File: build.gradle.kts (hunk 1 above)
Comment (e5l) [reply build.gradle.kts:50]: Thanks, removed

File: build.gradle.kts [hunk 6]
```diff
@@ -27,66 +27,76 @@ plugins {
...
+
+apply(plugin = "kotlin-multiplatform")
+apply(plugin = "kotlinx-atomicfu")
+
+val coroutines_version: String by extra
+
+kotlin {
+    jvm {
+        compilations.all {
+            kotlinOptions.jvmTarget = "1.8"
```
[the remark this answers is not in the document]
Comment (e5l) [reply build.gradle.kts:47]: Done

**Pull request merged**